src/functions.rs

```diff
@@ -2969,19 +3026,58 @@ pub fn search_for_recursion(
     }
 
     if !functions.is_empty() {
-        let mut error: Option<CompileError> = None;
+        let mut loops: Vec<Vec<String>> = Vec::new();
 
         for func in functions {
-            if let Some(function_info) = function_map.get(func) {
-                error = Some(add_or_create_compile_error(
-                    error,
-                    "Recursive Function call found",
-                    function_info.declaration_file,
-                    function_info.get_declaration_range().unwrap_or_default(),
-                    "These calls have been found in a recursive loop.  There may be more than one recursive loop.",
-                ));
+            loops.push(find_recursion_loop(
+                func,
+                function_map,
+                terminated_list,
+                &mut Vec::new(),
+            )?);
+        }
+
+        // Now we find the smallest loop and tell that to the user
+        let mut smallest: i32 = -1;
```

Comment:
This should be simpler and more readable with min_by https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.min_by